Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentSchema.h

```diff
@@ -312,6 +312,74 @@ namespace AZ::DocumentPropertyEditor
         AZStd::optional<EnumValuesContainer> DomToValue(const Dom::Value& value) const override;
     };
 
+    template<typename GenericValueType>
+    class GenericValueListAttributeDefinition final
+        : public AttributeDefinition<AZStd::vector<AZStd::pair<GenericValueType, AZStd::string>>>
```

Comment:
Perhaps alias this bit: `AZStd::vector<AZStd::pair<GenericValueType, AZStd::string>>` it's difficult to read

Reply:
I was originally aliasing this stuff before the `EnumValue` PR, but when we have to use templating for the AttributeDefinitions that need to support multiple types (int, unsigned int, etc) the alias won't work. It's because the alias would need the template as well.

If there is a way to do that then I'll definitely get this changed here, and for similar types.

Reply:
I'll check this out in the morning and update the PR if I can get it working.

Reply:
I tried the approach above but the code gets more complicated and difficult to read. If this were lower level code then I think it would look normal (it's often harder to read) but I think I'll leave this alone for now.

Reply:
You could go with an alias called ComboBoxContainer for the AZStd::vector
```c++
template <typename T>
using ComboBoxContainer = AZStd::vector<AZStd::pair<T, AZStd::string>;
```